account_payment_method_or_mode_fs_storage/models/account_payment_order.py

```diff
@@ -12,12 +12,17 @@ class AccountPaymentOrder(models.Model):
 
     _inherit = "account.payment.order"
 
+    def _get_storage(self):
+        if self.env.company.fs_storage_source_payment == "mode":
+            return self.payment_mode_id.storage
+        return self.payment_method_id.storage
```

Comment:
IMO you should do a `else` section.
And if doesn't fit to your case: `return ''` (in case of another module extend your `fs_storage_source_payment`.